src/CommunityToolkit.Maui/Layouts/StateLayout/StateLayoutController.shared.cs

```diff
@@ -0,0 +1,326 @@
+﻿using CommunityToolkit.Maui.Core;
+using StackLayout = Microsoft.Maui.Controls.StackLayout;
+using ViewExtensions = Microsoft.Maui.Controls.ViewExtensions;
+
+namespace CommunityToolkit.Maui.Layouts;
+
+/// <summary>
+/// StateLayout Controller
+/// </summary>
+public class StateLayoutController: IDisposable
+{
+	readonly WeakReference<Layout> layoutWeakReference;
+	bool layoutIsGrid;
+	LayoutState previousState = LayoutState.None;
+	IList<View> originalContent = Enumerable.Empty<View>().ToList();
```

Comment:
This is a `private` field and we just use `List<T>` during the interaction, so we don't need to use the interface here.

```suggestion
	List<View> originalContent = Enumerable.Empty<View>().ToList();
```